### DISCUSSION TITLE
Refactoring of package causes inconsistent state after poetry update.

### DISCUSSION THREAD

**Discussion by Kn0rk** [python-poetry/poetry#7223] [General]
So I am devolving a package with some meta/config files that are distributed as part of the package. Lets say the pkg looks something like this:

```
packagename
   Old-config.conf
   mycode.py
```

and I now refactored it to:

```
packagename
   metadata
      new-location.conf
   mycode.py
```

Lets assume we have a client with the old version we want to update. That means the old file structure will be present in the site-packages. When I know run a poetry update it keeps these old files in place, updates files that are still present and in the same location and adds the now new files. As a result both new and old version of the meta-data is now present. In my case this causes outdated meta-data to hang around after an update breaking my package.

Is there a way to just completely remove the package and reinstall from scratch similar to `pips --force-reinstall `flag. How do you handle this? Or do you handle refactoring of packages?

**Comment by finswimmer** [comment]
How do you install the package at the client?

**Reply by Kn0rk** [reply]
The package has been added at the client with

`poetry add package`

then i refactor the package on the devolper end, publish it and and call

`poetry update package`

On the client side.

**Reply by finswimmer** [reply]
Hm, looks good to me. Have you changed the version number of the package after developing?

**Reply by Kn0rk** [reply]
Yes but only a minor version. So 0.0.1 to 0.0.2
